### PR TITLE
Iir1 bump cci

### PR DESCRIPTION
**Pull request by wouterz**
Specify library name and version:  **iir1**

bumping with a cci version as I have no indication when a new version is to be released.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by wouterz** [comment]
A local build with the same profile seems to succeed, not sure where this is coming from
```
...
[HOOK - conan-center.py] post_package(): [MS RUNTIME FILES (KB-H021)] OK
iir1/cci.20211227 package(): Packaged 1 '.dll' file: iir.dll
iir1/cci.20211227 package(): Packaged 14 '.h' files
iir1/cci.20211227 package(): Packaged 1 '.lib' file: iir.lib
iir1/cci.20211227 package(): Packaged 1 file: COPYING
iir1/cci.20211227: Package 'd32867aa59c021b5efc1f4fb5b88d63569607fe1' created
...
```

File: recipes/iir1/all/conandata.yml [hunk 1]
```diff
@@ -2,8 +2,6 @@ sources:
   "1.9.0":
     url: "https://github.com/berndporr/iir1/archive/refs/tags/1.9.0.tar.gz"
     sha256: "BF2C3CD819151D5B85E84CC8349C1AA9DD5E4157A7070BDD143130278B4375E8"
-patches:
-  "1.9.0":
-    - patch_file: "patches/0002-Add-runtime-export-to-install.patch"
-      base_path: "source_subfolder"
-
```
Comment (ericLemanissier) [inline recipes/iir1/all/conandata.yml:9]: why did you remove this ?

**Review by ericLemanissier — COMMENTED** [review]
this could be the cause of the failure

File: recipes/iir1/all/conanfile.py [hunk 2]
```diff
@@ -72,13 +76,15 @@ def _configure_cmake(self):
...
         self._cmake = CMake(self)
+        self._cmake.definitions['IIR1_NO_EXCEPTIONS'] = self.options.noexceptions
         self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
     def build(self):
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
             tools.patch(**patch)
         cmake = self._configure_cmake()
+
```
Comment (ericLemanissier) [inline recipes/iir1/all/conanfile.py:87]:
```suggestion
```

File: recipes/iir1/all/conandata.yml (hunk 1 above)
Comment (wouterz) [reply recipes/iir1/all/conandata.yml]: Oops that was an accident. Also just realized I didnt read the log right, as the error happened in `1.9.0` and not the new cci. So this might very well be it

File: recipes/iir1/all/conanfile.py [hunk 3]
```diff
@@ -50,6 +52,8 @@ def export_sources(self):
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if self.version == "1.9.0" and self.options.noexceptions:
+            raise ConanInvalidConfiguration("1.9.0 does not the option to disable exceptions")
```
Comment (SSE4) [inline recipes/iir1/all/conanfile.py:56]:
```suggestion
            raise ConanInvalidConfiguration("1.9.0 does not have the option to disable exceptions")
```

**Review by SSE4 — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Comment by wouterz** [comment]
I got away with comparing strings, but I wondered if there is a way to check the chronological order of versions, to check before or after specific versions? Parsing into major.minor.patch would work for normal versions but not custom release naming.

**Review by ericLemanissier — DISMISSED** [review]

**Comment by wouterz** [comment]
Any logs on the unexpected failure?

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
Please do not force push 🙏 GitHub forces us to restart the review which is not fun!

**Comment by wouterz** [comment]
> Please do not force push 🙏 GitHub forces us to restart the review which is not fun!

Sorry, I pushed with wrong git config and required a new CLA, so I decided to remove that commit and redo it

**Comment by prince-chrismc** [comment]
Oh then you had no choice, that's absolutely alright :)

File: recipes/iir1/all/conanfile.py [hunk 4]
```diff
@@ -72,6 +76,7 @@ def _configure_cmake(self):
             return self._cmake
 
         self._cmake = CMake(self)
+        self._cmake.definitions['IIR1_NO_EXCEPTIONS'] = self.options.noexceptions
```
Comment (uilianries) [inline recipes/iir1/all/conanfile.py:79]:
```suggestion
        self._cmake.definitions['IIR1_NO_EXCEPTIONS'] = self.options.get_safe("noexceptions", False)
```
You still can use for any version. CMake will show a warning as unused definition, but it does not matter.

File: recipes/iir1/all/conanfile.py [hunk 5]
```diff
@@ -108,3 +113,7 @@ def package_info(self):
         self.cpp_info.components["iir"].set_property("cmake_target_name", "iir::{}".format(name))
 
         self.cpp_info.components["iir"].libs = [name]
+
+        if self.options.noexceptions:
```
Comment (uilianries) [inline recipes/iir1/all/conanfile.py:117]:
```suggestion
        if self.options.get_safe("noexceptions", False):
```

**Review by SSE4 — APPROVED** [review]

**Comment by ericLemanissier** [comment]
[1.9.1](https://github.com/berndporr/iir1/releases/tag/1.9.1) is now out

**Review by jgsogo — CHANGES_REQUESTED** [review]

File: recipes/iir1/all/conanfile.py [hunk 6]
```diff
@@ -50,6 +52,8 @@ def export_sources(self):
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if self.version == "1.9.0":
```
Comment (jgsogo) [inline recipes/iir1/all/conanfile.py:55]: Probably this should be 
```suggestion
        if tools.Version(self.version) < "1.9.1":
```

Comment (wouterz) [reply recipes/iir1/all/conanfile.py:55]: Yes! This is what I wanted to ask in my previous comment, as using self.version is clearly a string comparison I was curious how to do this correctly for multiple versions. tools.Version is the answer :)

**Review by jgsogo — APPROVED** [review]
👍
